### ISSUE TITLE
QuickFixTest1d8.testRemoveRedundantNonNull fails since I20230202-1800

### ISSUE THREAD

**Issue by iloveeclipse** [eclipse-jdt/eclipse.jdt.ui#423]
QuickFixTest1d8.testRemoveRedundantNonNull fails since I20230202-1800.
```
Wrong number of problems, is: 2, expected: 1 Pb(922) The nullness annotation is redundant with a default that applies to this location[91 ,105] Pb(922) The nullness annotation is redundant with a default that applies to this location[91 ,105]

java.lang.AssertionError: Wrong number of problems, is: 2, expected: 1
Pb(922) The nullness annotation is redundant with a default that applies to this location[91 ,105]
Pb(922) The nullness annotation is redundant with a default that applies to this location[91 ,105]

at org.junit.Assert.fail(Assert.java:89)
at org.eclipse.jdt.ui.tests.quickfix.QuickFixTest.assertNumberOfProblems(QuickFixTest.java:254)
at org.eclipse.jdt.ui.tests.quickfix.QuickFixTest.collectCorrections(QuickFixTest.java:229)
at org.eclipse.jdt.ui.tests.quickfix.QuickFixTest.collectCorrections(QuickFixTest.java:224)
at org.eclipse.jdt.ui.tests.quickfix.QuickFixTest1d8.testRemoveRedundantNonNull(QuickFixTest1d8.java:1743)
```

Regression from https://github.com/eclipse-jdt/eclipse.jdt.core/pull/683

**Comment by iloveeclipse** [comment]
@stephan-herrmann : I've pushed https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/425 in case it would take longer to fix JDT core problem, so that JDT UI builds can succeed in Jenkins (right now they all fail due the broken test).

Just give a hint if that should be merged or not, I will wait with merging for now, especially ECA doesn't work anymore today: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/2635

**Comment by stephan-herrmann** [comment]
Thanks for the heads-up. Of course no warning should be reported twice.

I'll look into it later today. If you need to fix JDT/UI right now, feel free to merge #425 - I'd revert that when the issue in JDT/Core is fixed.

**Comment by iloveeclipse** [comment]
@stephan-herrmann : I will wait, no one can merge anyway now. 
In best case you will have a fix in near future and we can close #425 without merging.

**Comment by stephan-herrmann** [comment]
@iloveeclipse if you see  eclipse-jdt/eclipse.jdt.core#707 succeed, before I get online again, feel free to merge it (although it stirred up some more dust, which I'll clean up later).
